src/padding.jl

```diff
@@ -7,7 +7,7 @@ export pad_constant, pad_repeat, pad_reflect, pad_zeros
 Pad the array `x` with zeros.
 Equivalent to [`pad_constant`](@ref) with the constant equal to 0. 
 """
-pad_zeros(x::AbstractArray, pad::NTuple{M,Int}; dims = 1:M÷2) where M =
+pad_zeros(x::AbstractArray, pad; dims = :) where M =
```

Comment:
There is an extra `where M`.
Also, when doing `pad_zeros(x, 2)`, I agree that in a generic context `dims = :` would be the correct default, but for NN `dims = 1:ndims(x)-3` is the one that makes most sense and it is going to correspond to what the flux's layer `PadZeros(2)`  will do. I don't dislike `:`, just saying we should be careful and consider the NN usage and interaction with Flux

Reply:
I wouldn't want an extra Pad* layer. There's no value to it.